controllers/controller.js

```diff
@@ -0,0 +1,49 @@
+const TaskModel = require('../models/task');
+
+// find all tasks in the database
+module.exports.findAllTasks = async (req, res) => {
+  const tasks = await TaskModel.find();
+  res.json(tasks);
+}
+
+// add new task to database
+module.exports.addTask = async (req, res) => {
+  const taskToBeAdded = new TaskModel(req.body);
+  const tasks = await taskToBeAdded.save();
+  res.json(tasks);
+}
+
+// update title of the tasks
+module.exports.updateTitle = async (req, res) => {
+  await TaskModel.findById(req.params.id, (err, tasks) => {
+    if (err) {
+      return res.status(500).json(err);
+    };
+    tasks.title = req.body.title;
+    tasks.save();
+    return res.json(tasks);
+  });
+}
+
+// update status of the tasks
+module.exports.updateStatus = async (req, res) => {
+  await TaskModel.findById(req.params.id, (err, tasks) => {
+    if (err) {
+      return res.status(500).json(err);
+    };
+    tasks.title = req.body.isCompleted;
+    tasks.save();
+    return res.json(tasks);
+  });
+}
```

Comment:
Why did you have to split the update into 2 funtions?

Reply:
On the example website, I see user can only update 1 thing at a time:
1. Update status by a checkbox
2. Update the title by clicking on the title
So I think there will be 2 function which do each of above task

Reply:
I think we should use only one update function and using PUT method instead of PATCH. Therefore, you can update both title and status of the task with one end-point.

Reply:
For example, using the API `findByIdAndUpdate` of mongoose
```
module.exports.updateTask = async (req, res) => {
  await TaskModel.findByIdAndUpdate(req.param.id, (err, task) => {
    if (err) return res.status(500).json(err)
    
    return res.json(task);
  }
}
```
Same for delete task
Check the Mongoose API here: https://mongoosejs.com/docs/api.html